[upstream_ut]  NotImplementedError: The operator 'aten::_scaled_dot_product_efficient_attention_backward' is not c

Cases:
op_ut,third_party.torch-xpu-ops.test.xpu.test_transformers_xpu.TestSDPAFailureModesXPU,test_mem_eff_attention_backward_requires_bias_when_bias_grad_xpu

pytest_command:
cd <pytorch>/third_party/torch-xpu-ops/test/xpu  && PYTORCH_TEST_WITH_SLOW=1 pytest -v third_party/torch-xpu-ops/test/xpu/test_transformers_xpu.py -k test_mem_eff_attention_backward_requires_bias_when_bias_grad_xpu

Error Message:
 NotImplementedError: The operator 'aten::_scaled_dot_product_efficient_attention_backward' is not currently implemented for the XPU device. Please open a feature on https://github.com/intel/torch-xpu-ops/issues. You can set the environment variable `PYTORCH_ENABLE_XPU_FALLBACK=1` to use the CPU implementation as a fallback for XPU unimplemented operators. WARNING: this will bring unexpected performance compared with running natively on XPU. ; AssertionError: "bias_requires_grad is true but no bias was provided" does not match "The operator 'aten::_scaled_dot_product_efficient_attention_backward' is not currently implemented for the XPU device. Please open a feature on https://github.com/intel/torch-xpu-ops/issues. You can set the environment variable `PYTORCH_ENABLE_XPU_FALLBACK=1` to use the CPU implementation as a fallback for XPU unimplemented operators. WARNING: this will bring unexpected performance compared with running natively on XPU." 

Trace Example:

```
Command: cd <pytorch>/third_party/torch-xpu-ops/test/xpu  && PYTORCH_TEST_WITH_SLOW=1 pytest -v third_party/torch-xpu-ops/test/xpu/test_transformers_xpu.py -k test_mem_eff_attention_backward_requires_bias_when_bias_grad_xpu
NotImplementedError: The operator 'aten::_scaled_dot_product_efficient_attention_backward' is not currently implemented for the XPU device. Please open a feature on https://github.com/intel/torch-xpu-ops/issues. You can set the environment variable `PYTORCH_ENABLE_XPU_FALLBACK=1` to use the CPU implementation as a fallback for XPU unimplemented operators. WARNING: this will bring unexpected performance compared with running natively on XPU.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/daisyden/upstream/test_transformers/third_party/torch-xpu-ops/test/xpu/test_transformers_xpu.py", line 2063, in test_mem_eff_attention_backward_requires_bias_when_bias_grad
    with self.assertRaisesRegex(RuntimeError, error_str):
  File "/home/daisyden/miniforge3/envs/test_transformers/lib/python3.10/unittest/case.py", line 239, in __exit__
    self._raiseFailure('"{}" does not match "{}"'.format(
  File "/home/daisyden/miniforge3/envs/test_transformers/lib/python3.10/unittest/case.py", line 163, in _raiseFailure
    raise self.test_case.failureException(msg)
AssertionError: "bias_requires_grad is true but no bias was provided" does not match "The operator 'aten::_scaled_dot_product_efficient_attention_backward' is not currently implemented for the XPU device. Please open a feature on https://github.com/intel/torch-xpu-ops/issues. You can set the environment variable `PYTORCH_ENABLE_XPU_FALLBACK=1` to use the CPU implementation as a fallback for XPU unimplemented operators. WARNING: this will bring unexpected performance compared with running natively on XPU."

To execute this test, run the following from the base repo dir:
    python test/xpu/test_transformers_xpu.py TestSDPAFailureModesXPU.test_mem_eff_attention_backward_requires_bias_when_bias_grad_xpu

This message can be suppressed by setting PYTORCH_PRINT_REPRO_ON_FAILURE=0
```

--- a/src/ATen/native/transformers/Attention.cpp
+++ b/src/ATen/native/transformers/Attention.cpp
@@ -520,6 +520,10 @@ _scaled_dot_product_efficient_attention_backward_xpu(
   if (grad_input_mask[2])
     v.requires_grad_(true);
 
+  TORCH_CHECK(
+      !grad_input_mask[3] || attn_bias.defined(),
+      "bias_requires_grad is true but no bias was provided");
+
   std::optional<Tensor> attn_bias_opt;
   Tensor ab;
   if (attn_bias.defined()) {

